tests/test_mbed_targets.py

```diff
@@ -67,55 +67,69 @@ def test_cannot_compare_with_non_mbed_target_instance(self):
         target = _make_mbed_target(product_code="1000")
         self.assertFalse(target == "1000")
 
-    def test_hash_is_equal_to_hash_of_product_code(self):
-        tgt = _make_mbed_target(product_code="0100")
-        self.assertEqual(tgt.__hash__(), hash(tgt.product_code))
+    def test_hash_is_equal_to_hash_of_target_properties(self):
+        tgt = _make_mbed_target(product_code="0100", platform_name="a", board_type="b")
+        self.assertEqual(tgt.__hash__(), hash(tgt.product_code) ^ hash(tgt.platform_name) ^ hash(tgt.board_type))
```

Comment:
Is `tgt.__hash__()` equivalent to `hash(tgt)`? Might be better to reduce the dunders. :)